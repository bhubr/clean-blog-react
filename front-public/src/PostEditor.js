import React from 'react'

class PostEditor extends React.Component {
  constructor (props) {
    super(props)
    this.state = {
      title: '',
      content: ''
    }
    this.handleChange = this.handleChange.bind(this)
    this.handleSubmit = this.handleSubmit.bind(this)
  }
  handleChange (event) {
    this.setState({
      [event.target.name]: event.target.value
    })
  }
  handleSubmit (event) {
    event.preventDefault()
    this.props.addPost(this.state)
    this.setState({
      title: '',
      content: ''
    })
  }
  render () {
    const { title, content } = this.state
    return (
      <form onSubmit={ this.handleSubmit }>
        <input name="title" placeholder="Post title" value={ title } onChange={ this.handleChange } /><br />
        <textarea name="content" placeholder="Post content" value={ content } onChange={ this.handleChange }></textarea><br />
        <button type="submit">Publish</button>
      </form>
    )
  }
}

export default PostEditor
