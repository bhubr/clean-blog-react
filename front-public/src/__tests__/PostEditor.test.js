// PostEditor.test.js
import React from 'react'
import PostEditor from '../PostEditor'
import renderer from 'react-test-renderer'

test('Check that output is correct', () => {
  // Create mock function to use as "addPost" prop
  const mockAddPost = jest.fn()

  // Create component
  const component = renderer.create(
    <PostEditor addPost={mockAddPost} />
  )
  // Convert component to a tree of objects
  let tree = component.toJSON()
  // Match against snapshot
  expect(tree).toMatchSnapshot()

  const input = tree.children[0]
  // Pass a fake event to change event handler
  input.props.onChange({
    target: { name: 'title', value: 'a' }
  })
  tree = component.toJSON()
  expect(tree).toMatchSnapshot()

  const textarea = tree.children[2]
  // Pass a fake event to change event handler
  textarea.props.onChange({
    target: { name: 'content', value: 'z' }
  })
  tree = component.toJSON()
  expect(tree).toMatchSnapshot()

  tree.props.onSubmit({
    preventDefault: () => {}
  })
  tree = component.toJSON()
  expect(tree).toMatchSnapshot()

  expect(mockAddPost.mock.calls.length).toBe(1)
  expect(mockAddPost.mock.calls[0][0]).toEqual({
    title: 'a', content: 'z'
  })
})
