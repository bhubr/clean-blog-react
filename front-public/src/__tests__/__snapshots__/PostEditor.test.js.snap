// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Check that output is correct 1`] = `
<form
  onSubmit={[Function]}
>
  <input
    name="title"
    onChange={[Function]}
    placeholder="Post title"
    value=""
  />
  <br />
  <textarea
    name="content"
    onChange={[Function]}
    placeholder="Post content"
    value=""
  />
  <br />
  <button
    type="submit"
  >
    Publish
  </button>
</form>
`;

exports[`Check that output is correct 2`] = `
<form
  onSubmit={[Function]}
>
  <input
    name="title"
    onChange={[Function]}
    placeholder="Post title"
    value="a"
  />
  <br />
  <textarea
    name="content"
    onChange={[Function]}
    placeholder="Post content"
    value=""
  />
  <br />
  <button
    type="submit"
  >
    Publish
  </button>
</form>
`;

exports[`Check that output is correct 3`] = `
<form
  onSubmit={[Function]}
>
  <input
    name="title"
    onChange={[Function]}
    placeholder="Post title"
    value="a"
  />
  <br />
  <textarea
    name="content"
    onChange={[Function]}
    placeholder="Post content"
    value="z"
  />
  <br />
  <button
    type="submit"
  >
    Publish
  </button>
</form>
`;

exports[`Check that output is correct 4`] = `
<form
  onSubmit={[Function]}
>
  <input
    name="title"
    onChange={[Function]}
    placeholder="Post title"
    value=""
  />
  <br />
  <textarea
    name="content"
    onChange={[Function]}
    placeholder="Post content"
    value=""
  />
  <br />
  <button
    type="submit"
  >
    Publish
  </button>
</form>
`;
